tools/distpartitioning/data_shuffle.py

```diff
@@ -488,13 +494,15 @@ def exchange_feature(
     for idx in range(world_size):
         cond = partid_slice == (idx + local_part_id * world_size)
         gids_per_partid = gids_feat[cond]
-        tids_per_partid = tids_feat[cond]
         local_idx_partid = local_idx[cond]
 
         if gids_per_partid.shape[0] == 0:
             assert len(all_dims_dtype) % world_size == 0
             dim_len = int(len(all_dims_dtype) / world_size)
-            rank0_shape = tuple(list(np.zeros((dim_len - 1), dtype=np.int32)))
+            rank0_shape = list(np.zeros((dim_len - 1), dtype=np.int32))
+            # When a feature is 2-dim, the shape[1] (number of columns) should match the feature dimension.
+            if len(rank0_shape) == 2:
```

Comment:
`len(rank0_shape)` is expected to be `1` or `2`? or can be zero? add assertion here as well?

Reply:
Here's the way the (convoluted) way `rank0_shape` is derived, I believe it _should not_ be anything other than 1 or 2:

```
# featdata is described as "numpy array, features associated with this feature key being processed"
feat_dims_dtype = list(featdata_key.shape)
# Unless `featdata_key` was a 0-dimensional numpy array (which should not happen as that would be a scalar), 
# feat_dims_dtype will have length >= 1
The row `feat_dims_dtype.append(DATA_TYPE_ID[featdata_key.dtype])` incremements the length of `feat_dims_dtype` by 1, feat_dims_dtype now should have a length of >= 2

# The dims are gathered from all workers
all_dims_dtype = allgather_sizes(
        feat_dims_dtype, world_size, num_parts, return_sizes=True
    )
# all_dims_dtype has length num_workers*len(feat_dims_dtype)

# The later on:
assert len(all_dims_dtype) % world_size == 0
dim_len = int(len(all_dims_dtype) / world_size)
rank0_shape = list(np.zeros((dim_len - 1), dtype=np.int32))
# at this point, len(rank0_shape) will be equal to len(featdata_key.shape), which _should_ be 1 (e.g. labels) or 2 (multi-column feature)
```

I cannot think of way that the caller of this function, `exchange_features` would produce a zero-dimensional array as input to `exchange_feature`, or how a 3+ dimensional feature would be produced or handled.